### PR TITLE
Feature payment response

### LINKED ISSUE #174: Ticket purchase confirmation
Check all success/fail states on ticket purchase. Check what happens when leaving PG url.

### PR DESCRIPTION
**Pull request by PavelFilo**
closes #174 

- Payment fail states (for example leaving payment gateway)
- preventing going out of the payment gateway website
- wrapping invalid link states in ScreenView

TODO: in my next task update headers so I wont be able to go back from success page to gateway

### CODE REVIEW

File: app/(app)/(purchase-and-payment)/ticket-purchase.tsx [hunk 1]
```diff
@@ -49,26 +47,23 @@ const TicketPurchasePage = () => {
...
+      ) : data?.paymentStatus === 'SUCCESS' ? (
         <ContentWithAvatar
           variant="success"
           title={t('paymentSuccessful')}
           text={t('paymentSuccessfulText')}
         >
           <BoughtTicket ticket={data} />
         </ContentWithAvatar>
-      )}
+      ) : isError || data.paymentStatus === 'FAIL' ? (
```
Comment (radoslavzeman) [inline app/(app)/(purchase-and-payment)/ticket-purchase.tsx:60]: Please follow the standard order of states if possible:
https://tanstack.com/query/latest/docs/react/guides/queries#query-basics

File: app/(app)/(purchase-and-payment)/ticket-purchase.tsx [hunk 2]
```diff
@@ -19,27 +19,25 @@ export type TicketPurchaseSearchParams = {
...
-    // TODO check if ticket is paid and reset store
-    if (data) {
+    if (data?.paymentStatus === 'SUCCESS') {
       onPurchaseStoreUpdate(defaultInitialPurchaseStoreValues)
+    } else if (data?.paymentStatus === 'PENDING') {
+      refetch()
     }
-  }, [data, onPurchaseStoreUpdate])
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data])
```
Comment (radoslavzeman) [inline app/(app)/(purchase-and-payment)/ticket-purchase.tsx:40]: Does it make some troubles if also `onPurchaseStoreUpdate` is present in dependency array?

File: app/(app)/(purchase-and-payment)/purchase/payment.tsx [hunk 3]
```diff
@@ -10,24 +10,46 @@ export type PaymentSearchParams = {
...
+      <WebView
+        ref={webviewRef}
+        source={{ uri: paymentUrlDecoded }}
+        className="flex-1"
+        onNavigationStateChange={(e) => {
+          if (invalidPaymentGatewayLinks.some((url) => e.url.includes(url))) {
+            webviewRef.current?.stopLoading()
+            webviewRef.current?.goBack()
+          }
+        }}
```
Comment (radoslavzeman) [inline app/(app)/(purchase-and-payment)/purchase/payment.tsx:51]: Please explain in comment. It seems crucial for correct handling.

**Review by radoslavzeman — APPROVED** [review]

File: app/(app)/(purchase-and-payment)/ticket-purchase.tsx (hunk 2 above)
Comment (PavelFilo) [reply app/(app)/(purchase-and-payment)/ticket-purchase.tsx:40]: no, I only want to execute it when data changes and when data changes the newest version of `onPurchaseStoreUpdate` is used, so no need to add both... this is the reason i don't like that rule within useEffects :D

**Pull request merged**